pkg/envoy/ciliumenvoyconfig_test.go

```diff
@@ -892,3 +899,48 @@ func (s *JSONSuite) TestListenersAddedOrDeleted(c *C) {
 	res = new.ListenersAddedOrDeleted(&old)
 	c.Assert(res, Equals, true)
 }
+
+func TestGetEndpointsForLBBackends(t *testing.T) {
+	testAddr, err := netip.ParseAddr("192.128.1.1")
+	require.NoError(t, err)
+
+	serviceName := loadbalancer.ServiceName{
+		Namespace: "test-ns",
+		Name:      "test-name",
+		Cluster:   "test-cluster",
+	}
+	backends := map[string][]*loadbalancer.Backend{
+		"12000": {
+			{
+				L3n4Addr: *loadbalancer.NewL3n4Addr(loadbalancer.TCP, types.AddrClusterFrom(testAddr, 0), 12000, 3),
+			},
+		},
+		"13000": {
+			{
+				L3n4Addr: *loadbalancer.NewL3n4Addr(loadbalancer.TCP, types.AddrClusterFrom(testAddr, 0), 13000, 3),
+			},
+		},
+		"*": {
+			{
+				L3n4Addr: *loadbalancer.NewL3n4Addr(loadbalancer.TCP, types.AddrClusterFrom(testAddr, 0), 15000, 3),
+			},
+		},
+	}
+
+	endpoints := getEndpointsForLBBackends(serviceName, backends)
+	assert.Len(t, endpoints, 4)
+
+	var allClusterNames []string
+	for i := 0; i < len(endpoints); i++ {
```

Comment:
nit: could just be `for i := range endpoints`

Reply:
changed